configuration/pih/htmlforms/labResults.xml

```diff
@@ -175,19 +168,34 @@
                               placeholder="Enter Sample ID"/>
                     </field>
                 </span>
-            </p>
+            </div>
             <br/><br/>
-            <p>
+            <div>
                 <label for="test-date">
                     <uimessage code="pihcore.lab.test_date"/>
                 </label>
-                <span class="small">
-                    <field>
-                        <obs  id="test-date" conceptId="PIH:DATE OF LABORATORY TEST"
-                              placeholder="Enter date"/>
-                    </field>
-                </span>
-            </p>
+                <div class="small">
+                    <encounterDate id="encounter-date" default="now" />
+                    <div style="display: none;">
+                        <obs id="test-date" conceptId="PIH:DATE OF LABORATORY TEST" />
+                    </div>
+                </div>
+            </div>
+            <div>
+                <label for="specimen-collection-date">
+                    <uimessage code="pihcore.lab.specimen_collection_date"/>
+                </label>
+                <div class="small required">
+                    <obs id="specimen-collection-date"
+                         conceptId="PIH:SPUTUM COLLECTION DATE"
+                         allowTime="false" />
```

Comment:
It looks like you are adding a new obs here to store sputum collection date?  Previously, we had decided that specimen collection date would be tracked as the encounter date... for better or worse, as this form is in use, we can't change it without performing a data migration.  (fwiw, the idea was that the encounter date should get set to the date when the other obs in the encounter (ie the results) are clinically-relevant, which is the specimen collection data)

There's some general server-side validation that doesn't allow any of the obs value datetimes on a form to be after the encounter date, which is the source of the current error (this validation may not have been working previously which is why this is just coming up now).  We should disable the server-side validation for this, via the "allowFutureDates" feature, see: https://wiki.openmrs.org/display/docs/HTML+Form+Entry+Module+Reference#HTMLFormEntryModuleReference-%3Cobs%3E

Besides that, having JS logic to update the test date if necessary when the specimen date changes and vice versa makes sense to me,  thought  might be easier to just have a submission handler that validates that test date >= encounter date upon submission.  This is documented here, see "Defining javascript functions to call before form validation and submission":

https://wiki.openmrs.org/display/docs/HTML+Form+Entry+JavaScript+Reference

This not documented well/at all, but it's also possible to have the one-question-per-screen navigator jump back to a specific field when you display this.  I forget the details, but there are some API methods provided here, and it looks like there are some examples of usage within the form itself.

https://github.com/openmrs/openmrs-module-uicommons/blob/master/omod/src/main/webapp/resources/scripts/navigator/navigator.js#L119